app/src/main/java/com/example/ohthmhyh/HabitsFragment.java

```diff
@@ -49,16 +69,86 @@ public static HabitsFragment newInstance(String param1, String param2) {
+            v = LayoutInflater.from(getContext()).inflate(R.layout.alert_addhabit, null);
+            alertDialog.setView(v);
+
+            alertDialog.setTitle("Add a Habit");
+
+            EditText habitNameET = v.findViewById(R.id.enter_habit_name);
+            EditText habitDescriptionET = v.findViewById(R.id.enter_habit_des);
+            EditText habitDateET = v.findViewById(R.id.enter_date);
+
+            ToggleButton monFrequency = v.findViewById(R.id.mon);
+            ToggleButton tueFrequency = v.findViewById(R.id.tue);
+            ToggleButton wedFrequency = v.findViewById(R.id.wed);
+            ToggleButton thuFrequency = v.findViewById(R.id.thu);
+            ToggleButton friFrequency = v.findViewById(R.id.fri);
+            ToggleButton satFrequency = v.findViewById(R.id.sat);
+            ToggleButton sunFrequency = v.findViewById(R.id.sun);
+
+
+
+            alertDialog.setButton(AlertDialog.BUTTON_NEGATIVE, "Cancel",
+                    new DialogInterface.OnClickListener() {
+                        public void onClick(DialogInterface dialog, int which) {
+                            // Empty Listener as the Cancel button doesn't do anything.
+                        }
+                    });
+
+            alertDialog.setButton(AlertDialog.BUTTON_POSITIVE, "Add",
+                    new DialogInterface.OnClickListener() {
+                        public void onClick(DialogInterface dialog, int which) {
+                            habitName = habitNameET.getText().toString();
+                            habitDescription = habitDescriptionET.getText().toString();
+                            startDate = handleDate(habitDateET.getText().toString());
+
+                            if (monFrequency.isChecked()) schedule.add(Habit.Days.Mon);
+                            if (tueFrequency.isChecked()) schedule.add(Habit.Days.Tue);
+                            if (wedFrequency.isChecked()) schedule.add(Habit.Days.Wed);
+                            if (thuFrequency.isChecked()) schedule.add(Habit.Days.Thu);
+                            if (friFrequency.isChecked()) schedule.add(Habit.Days.Fri);
+                            if (satFrequency.isChecked()) schedule.add(Habit.Days.Sat);
+                            if (sunFrequency.isChecked()) schedule.add(Habit.Days.Sun);
+
+                            habitArrayList.add(new Habit(habitName, habitDescription, startDate, schedule));
+                        }
+                    });
+
+
+            alertDialog.show();
+        });
+
+        return view;
+    }
+
+
+    private LocalDate handleDate(String dateAsString) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d/MM/yyyy");
+
+        //convert String to LocalDate
+        LocalDate localDate = LocalDate.parse(dateAsString, formatter);
+        return localDate;
+    }
+
+    public ArrayList<Habit> getHabitArrayList() {
+        return habitArrayList;
     }
 }
```

Comment:
```suggestion
}

```

Reply:
I will work to complete those user stories tonight I hope.

Reply:
Awesome thanks @nguyentuthanhdn! I will merge this PR then and remove this branch. When you get those other user stories complete, feel free to create another PR.

Reply:
@nguyentuthanhdn, there has to be some changes in this PR to accommodate some of the changes that Josh made. I can be responsible for this, but the files you changed may look slightly different.